File: frontend/src/shared/types/hotel.ts
import { HotelStatus } from './api'

export interface Hotel {
  id: number
  owner_id: number
  name: string
  description: string
  address: string
  city: string
  country: string
  phone: string
  email: string
  star_rating: number
  status: HotelStatus
  rejection_reason?: string
  amenities: string
  created_at: string
}

export interface HotelSearchResult {
  id: number
  name: string
  description: string
  city: string
  country: string
  address: string
  star_rating: number
  amenities: string
  min_price: number
  available_rooms: number
  rooms: Room[]
}

export interface Room {
  id: number
  hotel_id: number
  name: string
  description: string
  room_type: string
  price_per_night: number
  capacity: number
  amenities: string
  is_available: boolean
  created_at: string
}

File: backend/models/hotel.py
from sqlmodel import SQLModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum


class HotelStatus(str, Enum):
    pending = "pending"
    approved = "approved"
    rejected = "rejected"


class Hotel(SQLModel, table=True):
    __tablename__ = "hotels"
    id: Optional[int] = Field(default=None, primary_key=True)
    owner_id: int = Field(foreign_key="users.id")
    name: str
    description: str = Field(default="")
    address: str
    city: str
    country: str
    phone: str = Field(default="")
    email: str = Field(default="")
    star_rating: int = Field(default=3, ge=1, le=5)
    status: HotelStatus = Field(default=HotelStatus.pending)
    rejection_reason: Optional[str] = None
    amenities: str = Field(default="[]")  # JSON string
    images: str = Field(default="[]")     # JSON string
    latitude: float = Field(default=0.0)
    longitude: float = Field(default=0.0)
    is_suspended: bool = Field(default=False)
    commission_rate: float = Field(default=0.10)
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

# Coordinates for map search (added as new columns)
# SQLite will add these on first run via create_all